tests/integration/pilot/istioctl_test.go

```diff
@@ -293,3 +293,79 @@ func jsonUnmarshallOrFail(t *testing.T, context, s string) interface{} {
 	}
 	return val
 }
+
+func TestProxyStatus(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
+				Prefix: "istioctl-ps",
+				Inject: true,
+			})
+
+			var a echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, echoConfig(ns, "a")).
+				BuildOrFail(ctx)
+
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+
+			podID, err := getPodID(a)
+			if err != nil {
+				ctx.Fatalf("Could not get Pod ID: %v", err)
+			}
+
+			var output string
+			var args []string
+			g := gomega.NewGomegaWithT(t)
+
+			// Ignore --namespace=dummy, we use fully qualified <pod>.<ns>
+			args = []string{"proxy-status"}
+			output, _ = istioCtl.InvokeOrFail(t, args)
+			// Just verify pod A is known to Pilot; implicitly this verifies that
+			// the printing code printed it.
+			g.Expect(output).To(gomega.ContainSubstring(fmt.Sprintf("%s.%s", podID, ns.Name())))
+
+			args = []string{"--namespace=dummy",
+				"proxy-status", fmt.Sprintf("%s.%s", podID, ns.Name())}
+			output, _ = istioCtl.InvokeOrFail(t, args)
+			g.Expect(output).To(gomega.ContainSubstring("Clusters Match"))
+			g.Expect(output).To(gomega.ContainSubstring("Listeners Match"))
+			g.Expect(output).To(gomega.ContainSubstring("Routes Match"))
+		})
+}
+
+func TestAuthZCheck(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
+				Prefix: "istioctl-authz",
+				Inject: true,
+			})
+
+			var a echo.Instance
```

Comment:
why not have a single pod shared between the tests then just run a serious of `istioctl` commands against it?

Reply:
I'd like to do that.  Is there a way that I can have multiple TestXYX() functions run after the same pod is ready?  Or is it expected that I have a single Test() function and call out to functions for each subcommand?

Reply:
Here is how you could do it (pseudocode, may be slightly off)

```
func TestIstioctl() {
  framework.NewTest.Run(func() {
    ns := ...
   echoInstance := setupInstance
   ctx.NewSubtest("describe").Run(func() { ... run istioctl test ...})
   ctx.NewSubtest("authz").Run(func() { ... run istioctl test ...})
}
```

Reply:
I believe these tests already run in parallel.   I like the look of that and will use that style in a follow-up.

Reply:
You can chose parallel or serial with Run or RunParallel I think. Parallel seems reasonable here